app/frontend/src/features/messages/requests/HostRequestSendField.tsx

```diff
@@ -149,6 +179,25 @@ export default function HostRequestSendField({
                 Cancel
               </FieldButton>
             )}
+            {hostRequest.status ===
+              HostRequestStatus.HOST_REQUEST_STATUS_CONFIRMED &&
+              availableRefrences &&
+              availableRefrences.availableWriteReferencesList.find(
+                ({ hostRequestId }) =>
+                  hostRequestId === hostRequest.hostRequestId
+              ) && (
```

Comment:
```suggestion
              availableRefrences?.availableWriteReferencesList.find(
                ({ hostRequestId }) =>
                  hostRequestId === hostRequest.hostRequestId
              ) && (
```

Reply:
Actually, this is identical to L143-147, so probably worth extracting to a variable called `canWriteReference` or something so we're not doing the same work twice